### PR TITLE
gates and buttons

### PR DESCRIPTION
**Pull request by RebeccaStankus**

### CODE REVIEW

File: DomainContent/Zombies/SOSButton.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
+//
+// SOSButton.js
+// 
+// Created by Rebecca Stankus on 03/07/2018
+// Copyright High Fidelity 2018
+//
+// Licensed under the Apache 2.0 License
+// See accompanying license file or http://apache.org/
+//
+/* global Pointers*/
```
Comment (elisa-lj11) [inline DomainContent/Zombies/SOSButton.js]: space after Pointers

File: DomainContent/Zombies/SOSButton.js [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+(function() {
+    var AUDIO_VOLUME_LEVEL = 0.2;
+    var DOWN_TIME_MS = 3000;
+    var DISABLED_TIME_MS = 10000;
+    var BOAT = "{4691d6ad-93f5-4456-90b9-95c9f2ef00b2}";
+    var BLOCK_BOAT_ACCESS = "{838f0103-1bfa-4c7c-a786-b3e8c0f71f8a}";
+    var BOAT_SOUND_POSITION = {
+        x:-4.8,
+        y:4.5,
+        z:-52
```
Comment (elisa-lj11) [inline DomainContent/Zombies/SOSButton.js]: space after colons

File: DomainContent/Zombies/SOSButton.js [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    var boatSound;
+
+    var Button = function() {
+        _this = this;
+    };
+
+    Button.prototype = {
+        preload: function(entityID) {
+            _this.entityID = entityID;
+            sound = SoundCache.getSound(Script.resolvePath("sounds/346108__limetoe__boat-horn.wav"));
```
Comment (elisa-lj11) [inline DomainContent/Zombies/SOSButton.js]: use variable for resolvePath for easier resetting of url if needed

File: DomainContent/Zombies/boatLeavingZone.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    var SURVIVOR_SCRIPT = "https://hifi-content.s3.amazonaws.com/davidback/development/zombies/zombieSurvivorScript.js";
+
+    var sound;
+
+    var BoatLeavingZone = function() {
+    };
+
+    BoatLeavingZone.prototype = {
+        preload: function(entityID) {
+            sound = SoundCache.getSound(Script.resolvePath("sounds/346108__limetoe__boat-horn.wav"));
```
Comment (elisa-lj11) [inline DomainContent/Zombies/boatLeavingZone.js]: same as above, variable for .wav

File: DomainContent/Zombies/boatServer.js [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+// Created by Rebecca Stankus on 03/07/2018
+// Copyright High Fidelity 2018
+//
+// Licensed under the Apache 2.0 License
+// See accompanying license file or http://apache.org/
+//
+
+(function() {
+    var BOAT_DOCKED_POSITION = {x:14.2899,y:-7.4687,z:36.7669};
+    // var BOAT_AWAY_POSITION = {x:-98.6815,y:-6.8897,z:150.2770};
```
Comment (elisa-lj11) [inline DomainContent/Zombies/boatServer.js]: remove extra comment if not needed

File: DomainContent/Zombies/boatServer.js [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+    var BOAT_RETURN_DOCKED_ROTATION = {x:1.0571,y:-143.6416,z:-2.4509};
+    var BOAT_MOVEMENT_INCREMENT_X = 0.01;
+    var BOAT_MOVEMENT_INCREMENT_Z = 0.01;
+    var BOAT_MOVEMENT_INTERVAL_MS = 10;
+    var AUDIO_VOLUME_LEVEL = 0.2;
+    var DEBUG = 1;
+    var TARGET_X_POSITION_AWAY = -36.2899;
+    // this can be rotated to move the boat's light
+    // var BOAT_LIGHT_ORIGIN = "{242b05c3-5367-4aa2-bb94-63255a6b4b71}";
+    // to turn right, x goes down, y goes down, and z goes up
```
Comment (elisa-lj11) [inline DomainContent/Zombies/boatServer.js]: remove these extra comments

File: DomainContent/Zombies/boatServer.js [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+
+    var _this;
+    var currentPosition = JSON.parse(JSON.stringify(BOAT_DOCKED_POSITION));
+    // var currentRotation = JSON.parse(JSON.stringify(BOAT_INITIAL_DOCKED_ROTATION));
```
Comment (elisa-lj11) [inline DomainContent/Zombies/boatServer.js]: remove commented out code

File: DomainContent/Zombies/boatServer.js [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+    var boatSound;
+    var boatSoundInjector;
+    var hasLeft = false;
+    var returning = false;
+    var moving;
+    // var turning;
```
Comment (elisa-lj11) [inline DomainContent/Zombies/boatServer.js]: remove

File: DomainContent/Zombies/boatServer.js [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+    };
+
+    Boat.prototype = {
+        remotelyCallable: ['reset', 'horn', 'leaveIsland', 'approachIsland'],
+        preload: function(entityID) {
+            if (DEBUG) {
+                print("hello boat server script with movement print");
+            }
+            _this.entityID = entityID;
+            boatSound = SoundCache.getSound(Script.resolvePath("sounds/346108__limetoe__boat-horn.wav"));
```
Comment (elisa-lj11) [inline DomainContent/Zombies/boatServer.js]: .wav variable

File: DomainContent/Zombies/boatServer.js [hunk 10]
```diff
@@ -0,0 +1,168 @@
...
+            }
+            Entities.editEntity(_this.entityID, {
+                position: BOAT_DOCKED_POSITION,
+                rotation: Quat.fromPitchYawRollDegrees(BOAT_INITIAL_DOCKED_ROTATION.x,BOAT_INITIAL_DOCKED_ROTATION.y,
+                    BOAT_INITIAL_DOCKED_ROTATION.z)
+            });
+            print("position from getEntityProperties is " + JSON.stringify(Entities.getEntityProperties(_this.entityID, 
+                'position').position));
+        },
+        horn: function(){
```
Comment (elisa-lj11) [inline DomainContent/Zombies/boatServer.js]: space between parantheses and bracket

File: DomainContent/Zombies/boatServer.js [hunk 11]
```diff
@@ -0,0 +1,168 @@
...
+        },
+        approachIsland: function() {
+            if (DEBUG) {
+                print("approachIsland method");
+            }
+            if (DEBUG) {
+                print("calling turnAround");
+            }
+            _this.turnAround();
+            if ( hasLeft && !returning) {
```
Comment (elisa-lj11) [inline DomainContent/Zombies/boatServer.js]: delete space before `hasLeft`

File: DomainContent/Zombies/boatServer.js [hunk 12]
```diff
@@ -0,0 +1,168 @@
...
+                        });
+                    } else {
+                        if (moving) {
+                            Script.clearInterval(moving);
+                        }
+                    }
+                }, BOAT_MOVEMENT_INTERVAL_MS);
+                
+                
+                // make this a triangle or circle to get it in position/rotation to head back to the dock
```
Comment (elisa-lj11) [inline DomainContent/Zombies/boatServer.js]: remove comment

File: DomainContent/Zombies/boatServer.js [hunk 13]
```diff
@@ -0,0 +1,168 @@
...
+                boatSoundInjector.stop();
+            }
+            if (moving) {
+                Script.clearInterval(moving);
+            }
+        }
+    };
+
+    return new Boat();
+});
```
Comment (elisa-lj11) [inline DomainContent/Zombies/boatServer.js]: newline

File: DomainContent/Zombies/busTriggersUnblockPathByCafe.js [hunk 14]
```diff
@@ -0,0 +1,14 @@
+(function() {
+    var AMBULANCE_BLOCKING_PATH_BY_CAFE = "{3f1e4501-622d-44c9-a353-7278a98b2142}";
+    print("bus zone");
```
Comment (elisa-lj11) [inline DomainContent/Zombies/busTriggersUnblockPathByCafe.js]: make debug optional

File: DomainContent/Zombies/busTriggersUnblockPathByCafe.js [hunk 15]
```diff
@@ -0,0 +1,14 @@
...
+    this.enterEntity = function() {
+        print("entered bus zone");
```
Comment (elisa-lj11) [inline DomainContent/Zombies/busTriggersUnblockPathByCafe.js]: make debug optional

File: DomainContent/Zombies/busTriggersUnblockPathByCafe.js [hunk 16]
```diff
@@ -0,0 +1,14 @@
...
+        if (Entities.getEntityProperties(AMBULANCE_BLOCKING_PATH_BY_CAFE,'visible').visible) {
+            print("unblock path");
```
Comment (elisa-lj11) [inline DomainContent/Zombies/busTriggersUnblockPathByCafe.js]: make debug optional

File: DomainContent/Zombies/busTriggersUnblockPathByCafe.js [hunk 17]
```diff
@@ -0,0 +1,14 @@
...
+            Entities.editEntity(AMBULANCE_BLOCKING_PATH_BY_CAFE, {
+                visible: false,
+                collisionless: true
+            });
+        }
+    };
+});
```
Comment (elisa-lj11) [inline DomainContent/Zombies/busTriggersUnblockPathByCafe.js]: newline

File: DomainContent/Zombies/cafeExplosionZone.js [hunk 18]
```diff
@@ -0,0 +1,44 @@
...
+        preload: function(entityID) {
+            sound = SoundCache.getSound(Script.resolvePath("sounds/156031__iwiploppenisse__explosion.wav"));
+            _this.ready = false;
+        },
+        enterEntity: function() {
+        // did explosion already happen?
+            var explosion = Entities.getEntityProperties(EXPLOSION_BY_CAFE, 'visible').visible;
+            if (!explosion && _this.ready) {
+                if (sound.downloaded) {
+                    print("sound!!!");
```
Comment (elisa-lj11) [inline DomainContent/Zombies/cafeExplosionZone.js]: debug optional

File: DomainContent/Zombies/cafeExplosionZone.js [hunk 19]
```diff
@@ -0,0 +1,44 @@
...
+        readyToExplode: function() {
+            _this.ready = true;
+        },
+        notReadyToExplode: function() {
+            _this.ready = false;
+        }
+    };
+
+    return new ExplosionZone();
+});
```
Comment (elisa-lj11) [inline DomainContent/Zombies/cafeExplosionZone.js]: newline

File: DomainContent/Zombies/gasMainButton.js [hunk 20]
```diff
@@ -0,0 +1,65 @@
...
+                    visible: false,
+                    collisionless: true,
+                    locked: true
+                });
+            }
+        }
+    };
+    
+    return new Button();
+});
```
Comment (elisa-lj11) [inline DomainContent/Zombies/gasMainButton.js]: newline

File: DomainContent/Zombies/gateButton.js [hunk 21]
```diff
@@ -0,0 +1,257 @@
...
+                return;
+            }
+        },
+        unload: function() {
+
+        }
+    };
+
+    return new Button();
+});
```
Comment (elisa-lj11) [inline DomainContent/Zombies/gateButton.js]: newline

File: DomainContent/Zombies/generatorButton.js [hunk 22]
```diff
@@ -0,0 +1,128 @@
...
+    var generatorSound;
+
+    var Button = function() {
+        _this = this;
+    };
+
+    Button.prototype = {
+        preload: function(entityID) {
+            _this.entityID = entityID;
+            sound = SoundCache.getSound(Script.resolvePath("sounds/259912__daroc__dieselgenerator.wav"));
```
Comment (elisa-lj11) [inline DomainContent/Zombies/generatorButton.js]: .wav variable

File: DomainContent/Zombies/generatorExplosionZone.js [hunk 23]
```diff
@@ -0,0 +1,34 @@
...
+    var AUDIO_VOLUME_LEVEL = 0.8;
+
+    var sound;
+
+    var ExplosionZone = function() {
+    };
+
+    ExplosionZone.prototype = {
+        preload: function(entityID) {
+            sound = SoundCache.getSound(Script.resolvePath("sounds/156031__iwiploppenisse__explosion.wav"));
```
Comment (elisa-lj11) [inline DomainContent/Zombies/generatorExplosionZone.js]: .wav variable

File: DomainContent/Zombies/generatorExplosionZone.js [hunk 24]
```diff
@@ -0,0 +1,34 @@
...
+                Entities.editEntity(EXPLOSION_BY_GENERATOR, {
+                    visible: true,
+                    collisionless: false
+                });
+            }
+        }
+    };
+
+    return new ExplosionZone();
+});
```
Comment (elisa-lj11) [inline DomainContent/Zombies/generatorExplosionZone.js]: newline

File: DomainContent/Zombies/models/Gate.mtl [hunk 25]
```diff
@@ -0,0 +1,8 @@
+newmtl Gate_mat
+illum 4
+Kd 0.50 0.50 0.50
+Ka 0.00 0.00 0.00
+Tf 1.00 1.00 1.00
+Ni 1.00
+
+map_Kd Gate_BaseColor.png
```
Comment (elisa-lj11) [inline DomainContent/Zombies/models/Gate.mtl]: newline

File: DomainContent/Zombies/resetZombieDomain.js [hunk 26]
```diff
@@ -0,0 +1,104 @@
...
+});
+/* Entities.editEntity(GATE5, {
+    position: GATE5_POSITION
+});
+Entities.editEntity(GATE6, {
+    position: GATE6_POSITION
+});
+Entities.editEntity(GATE7, {
+    position: GATE7_POSITION
+});*/
```
Comment (elisa-lj11) [inline DomainContent/Zombies/resetZombieDomain.js]: newline

File: DomainContent/Zombies/resetZombieDomain.js [hunk 27]
```diff
@@ -0,0 +1,104 @@
...
+Entities.editEntity(GATE2, {
+    position: GATE2_POSITION
+});
+Entities.editEntity(GATE3, {
+    position: GATE3_POSITION
+});
+Entities.editEntity(GATE4, {
+    position: GATE4_POSITION
+});
+/* Entities.editEntity(GATE5, {
```
Comment (elisa-lj11) [inline DomainContent/Zombies/resetZombieDomain.js]: take out commented code

File: DomainContent/Zombies/resetZombieDomain.js [hunk 28]
```diff
@@ -0,0 +1,104 @@
...
+};
+
+var GATE4 = "{a71682be-bef9-477e-89e1-2c8bee014965}";
+var GATE4_POSITION = {
+    x:-13.0005,
+    y:4.9248,
+    z:-52.6598
+};
+
+/* var GATE5 = "";
```
Comment (elisa-lj11) [inline DomainContent/Zombies/resetZombieDomain.js]: take out commented code

File: DomainContent/Zombies/models/Gate.mtl (hunk 25 above)
Comment (RebeccaStankus) [reply DomainContent/Zombies/models/Gate.mtl]: Is that necessary on mtl files?

File: DomainContent/Zombies/SOSButton.js [hunk 29]
```diff
@@ -0,0 +1,156 @@
...
+//
+/* global Pointers */
+
+(function() {
+    var AUDIO_VOLUME_LEVEL = 0.2;
+    var DOWN_TIME_MS = 3000;
+    var DISABLED_TIME_MS = 10000;
+    var BOAT = "{4691d6ad-93f5-4456-90b9-95c9f2ef00b2}";
+    var BLOCK_BOAT_ACCESS = "{838f0103-1bfa-4c7c-a786-b3e8c0f71f8a}";
+    var BOAT_HORN = "sounds/346108__limetoe__boat-horn.wav";
```
Comment (elisa-lj11) [inline DomainContent/Zombies/SOSButton.js]: rename to BOAT_HORN_SOUND

File: DomainContent/Zombies/boatLeavingZone.js [hunk 30]
```diff
@@ -0,0 +1,60 @@
...
+// See accompanying license file or http://apache.org/
+//
+
+(function () {
+    var BOAT = "{4691d6ad-93f5-4456-90b9-95c9f2ef00b2}";
+    var DEBUG = 0;
+    var AUDIO_VOLUME_LEVEL = 0.5;
+    var NEGATIVE = -1;
+    var SURVIVOR_SCRIPT = "https://hifi-content.s3.amazonaws.com/davidback/development/zombies/zombieSurvivorScript.js";
+    var BOAT_HORN = "sounds/346108__limetoe__boat-horn.wav";
```
Comment (elisa-lj11) [inline DomainContent/Zombies/boatLeavingZone.js]: same here

File: DomainContent/Zombies/models/Gate.mtl (hunk 25 above)
Comment (elisa-lj11) [reply DomainContent/Zombies/models/Gate.mtl]: unsure but github is unhappy :(

**Pull request merged**